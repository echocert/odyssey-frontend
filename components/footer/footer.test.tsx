import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import '@testing-library/jest-dom';
import { create } from "react-test-renderer";
import Footer from './footer';


describe('Footer Section', () => {
  it('renders the Footer Screen', () => {
    render(<Footer />)

    const footer = screen.getByTestId('footer');
    expect(footer).toBeInTheDocument()
  })
  it('it check if the Logo is rendered', () => {
    render(<Footer />)

    const logo = screen.getByTestId('footer-logo');
    expect(logo).toBeInTheDocument()
  })
  it('check if the Youtube Link is rendered', () => {
    render(<Footer />)

    const youtube_link = screen.getByTestId('youtube-link');
    expect(youtube_link).toBeInTheDocument()
    userEvent.click(youtube_link)
  })
  it('check if the Instagram Link is rendered', () => {
    render(<Footer />)

    const instagram_link = screen.getByTestId('instagram-link');
    expect(instagram_link).toBeInTheDocument()
    userEvent.click(instagram_link);
  })
  it('check if the Facebook Link is rendered', () => {
    render(<Footer />)

    const facebook_link = screen.getByTestId('facebook-link');
    expect(facebook_link).toBeInTheDocument()
    userEvent.click(facebook_link);
  })
  it('check if the Linkedin Link is rendered', () => {
    render(<Footer />)
    const linkedin_link = screen.getByTestId('linkedin-link');
    expect(linkedin_link).toBeInTheDocument()
    userEvent.click(linkedin_link);
  })
  it('check if the Conditions Terms is rendered', () => {
    render(<Footer />)

    const terms_conditions = screen.getByTestId('terms-conditions');
    expect(terms_conditions).toBeInTheDocument()
    userEvent.click(terms_conditions)
  })
  it("Verify the snapshot of Footer section is matched", () => {
    const wrap = create(<Footer />).toJSON();
    expect(wrap).toMatchSnapshot();
  });
})