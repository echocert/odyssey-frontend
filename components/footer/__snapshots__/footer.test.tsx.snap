// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Section Verify the snapshot of Footer section is matched 1`] = `
<div
  className="footer__wrapper"
>
  <div
    className="footer-container"
    data-testid="footer"
  >
    <div
      className="footer-container__item"
      style={
        {
          "gridArea": "item1",
        }
      }
    >
      <img
        alt=""
        className="footer__logo"
        data-testid="footer-logo"
        src="../assets/images/odyssey_logo_email.jpg"
      />
    </div>
    <div
      className="footer-container__item"
      style={
        {
          "gridArea": "item2",
          "placeItems": "center",
        }
      }
    >
      <label>
        ORC Solution
      </label>
      <label>
        PO BOX 391203
      </label>
      <label>
        Anza, California, USA
      </label>
      <label>
        92539
      </label>
      <label>
        650.249.5777
      </label>
      <label>
        info@orcsolution.com
      </label>
    </div>
    <div
      className="footer-container__item"
      style={
        {
          "gridArea": "item4",
        }
      }
    >
      <label>
        <a
          className="terms three"
          data-testid="terms-conditions"
          href="/terms-and-conditions"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onTouchStart={[Function]}
        >
          Terms and Conditions
        </a>
      </label>
      <label>
        <a
          className="terms three"
          href="/privacy-policy"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onTouchStart={[Function]}
        >
          Privacy policy
        </a>
      </label>
    </div>
    <div
      className="footer-container__item"
      style={
        {
          "gridArea": "item5",
        }
      }
    >
      Copyright Â© 2024 ORC Solution | Powered by Odyssey
    </div>
    <div
      className="footer-container__item"
      style={
        {
          "gridArea": "item6",
        }
      }
    >
      <div
        style={
          {
            "display": "flex",
          }
        }
      >
        <div
          className="footer__social-icons"
          data-testid="youtube-link"
          onClick={[Function]}
        >
          <img
            alt=""
            src="../assets/images/youtube.png"
          />
        </div>
        <div
          className="footer__social-icons"
          data-testid="instagram-link"
          onClick={[Function]}
        >
          <img
            alt=""
            src="../assets/images/instagram.png"
          />
        </div>
        <div
          className="footer__social-icons"
          data-testid="facebook-link"
          onClick={[Function]}
        >
          <img
            alt=""
            src="../assets/images/facebook.png"
          />
        </div>
        <div
          className="footer__social-icons"
          data-testid="linkedin-link"
          onClick={[Function]}
        >
          <img
            alt=""
            src="../assets/images/linkedin.png"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
